### PR TITLE
androidenv: fix libgcc_s.so.1 -> not found

### PR DESCRIPTION
**Pull request by hadilq**
###### Description of problem
Fixes:
```
libgcc_s.so.1 -> not found!
```
which is reported in https://github.com/NixOS/nixpkgs/issues/226357. 
###### Changes
- adding `stdenv.cc.cc` dependency to `tools.nix` to fix the problem.
- also adding a missing line in the `pkgs/development/mobile/androidenv/README.md` file.
-  reviewed dependencies of `pkgs/development/mobile/androidenv/build-tools.nix`.

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [X] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/mobile/androidenv/README.md [hunk 1]
```diff
@@ -10,6 +10,7 @@ You may need to make yourself familiar with [tests](https://nixos.org/manual/nix
 
 ```shell
 $ export NIXPKGS_ALLOW_UNFREE=1
+$ export NIXPKGS_ACCEPT_ANDROID_SDK_LICENSE=1
```
Comment (amjoseph-nixpkgs) [inline pkgs/development/mobile/androidenv/README.md:13]: Could you please move this to a separate PR?  It is not really related to the `libgcc_s` issue and I'm not the right reviewer for it.

File: pkgs/development/mobile/androidenv/build-tools.nix [hunk 2]
```diff
@@ -4,7 +4,10 @@ deployAndroidPackage {
   inherit package os;
   nativeBuildInputs = [ makeWrapper ]
     ++ lib.optionals (os == "linux") [ autoPatchelfHook ];
-  buildInputs = lib.optionals (os == "linux") [ pkgs.glibc pkgs.zlib pkgs.ncurses5 pkgsi686Linux.glibc pkgsi686Linux.zlib pkgsi686Linux.ncurses5 pkgs.libcxx ];
+  buildInputs = lib.optionals (os == "linux") (
+    (with pkgs; [ glibc zlib ncurses5 stdenv.cc.cc ])
+    ++ (with pkgsi686Linux; [ glibc zlib ncurses5 ])
+  );
```
Comment (amjoseph-nixpkgs) [inline pkgs/development/mobile/androidenv/build-tools.nix:10]: Likewise; this looks like it's just a formatting change (I think?).  Please put it in a separate PR.

File: pkgs/development/mobile/androidenv/tools.nix [hunk 3]
```diff
@@ -6,7 +6,7 @@ deployAndroidPackage {
   nativeBuildInputs = [ makeWrapper ]
     ++ lib.optionals (os == "linux") [ autoPatchelfHook ];
   buildInputs = lib.optional (os == "linux") (
-      (with pkgs; [ glibc freetype fontconfig fontconfig.lib])
+      (with pkgs; [ glibc freetype fontconfig fontconfig.lib stdenv.cc.cc ])
```
Comment (amjoseph-nixpkgs) [inline pkgs/development/mobile/androidenv/tools.nix:9]: Is there a shorter reproducing expression than "[Try to rebuild with my system configuration](https://github.com/NixOS/nixpkgs/issues/226357)"?

I *think* you should be able to use `stdenv.cc.cc.libgcc or null` here, but I can't test it because I don't quite know how to reproduce the bug (without rebuilding all of @NotAShelf's NixOS install).

Comment (amjoseph-nixpkgs) [inline pkgs/development/mobile/androidenv/tools.nix:9]:
```suggestion
      (with pkgs; [ glibc freetype fontconfig fontconfig.lib
         stdenv.cc.cc # fix for https://github.com/NixOS/nixpkgs/issues/226357
      ])
```

**Review by ghost — COMMENTED** [review]
Looks good but most of this belongs in a separate PR.

If you can split that out and provide the `nix build` expression that reproduces the bug I will merge this.

File: pkgs/development/mobile/androidenv/tools.nix (hunk 3 above)
Comment (NotAShelf) [reply pkgs/development/mobile/androidenv/tools.nix:9]: I would love to be able to give you a shorter reproduction hint, but the only way I've been able to produce this is by building my system configuration with `diffoscope` in system packages.

File: pkgs/development/mobile/androidenv/README.md (hunk 1 above)
Comment (hadilq) [reply pkgs/development/mobile/androidenv/README.md]: Those changes was in the spirit of Boy Scout Rule:
> The Boy Scouts have a rule: “Always leave the campground cleaner than you found it”
but anyway! I revert them back as you requested. check https://github.com/NixOS/nixpkgs/pull/226489/commits/398f142623dee16188587d66c83375c5fddb739c

File: pkgs/development/mobile/androidenv/build-tools.nix (hunk 2 above)
Comment (hadilq) [reply pkgs/development/mobile/androidenv/build-tools.nix]: Those changes was in the spirit of Boy Scout Rule:

> The Boy Scouts have a rule: “Always leave the campground cleaner than you found it”

but anyway! I revert them back as you requested. check https://github.com/NixOS/nixpkgs/pull/226489/commits/398f142623dee16188587d66c83375c5fddb739c

File: pkgs/development/mobile/androidenv/tools.nix (hunk 3 above)
Comment (hadilq) [reply pkgs/development/mobile/androidenv/tools.nix]: applied in https://github.com/NixOS/nixpkgs/pull/226489/commits/575e9aad09b82b10e28d0efd34413b2c685481da

Comment (hadilq) [reply pkgs/development/mobile/androidenv/tools.nix]: I didn't build the `diffoscope` to reproduce it. I just ran the tests. Take a look at the [README.md](https://github.com/NixOS/nixpkgs/blob/e8121fa7db800faac2e7160b1958d7125c030073/pkgs/development/mobile/androidenv/README.md) file.

Comment (hadilq) [reply pkgs/development/mobile/androidenv/tools.nix]: `libgcc` also works. check out https://github.com/NixOS/nixpkgs/pull/226489/commits/0ed63be46aacdb79f19b003e25ec08a2e0ef5a61

File: pkgs/development/mobile/androidenv/tools.nix [hunk 4]
```diff
@@ -6,7 +6,9 @@ deployAndroidPackage {
   nativeBuildInputs = [ makeWrapper ]
     ++ lib.optionals (os == "linux") [ autoPatchelfHook ];
   buildInputs = lib.optional (os == "linux") (
-      (with pkgs; [ glibc freetype fontconfig fontconfig.lib])
+      (with pkgs; [ glibc freetype fontconfig fontconfig.lib
+        stdenv.cc.cc.libgcc # fix for https://github.com/NixOS/nixpkgs/issues/226357
```
Comment (amjoseph-nixpkgs) [inline pkgs/development/mobile/androidenv/tools.nix:10]:
```suggestion
        stdenv.cc.cc.libgcc or null # fix for https://github.com/NixOS/nixpkgs/issues/226357
```

Comment (hadilq) [reply pkgs/development/mobile/androidenv/tools.nix:10]: @github I'm in the Android app, and there's no apply button for the suggestion!

Anyway! I'll apply the suggestion as soon as possible, even though, I don't understand the reason behind the `or null` part. Would you explain more? In my view the `or null` is supposed to make it less likely to break in the future, right? But why do we want that when we can be explicit about the dependencies and have a fail-fast mindset to make reading, debugging, and fixing future problems easier?

Comment (Artturin) [reply pkgs/development/mobile/androidenv/tools.nix]: squashed and applied suggestion

**Comment by Artturin** [comment]
built diffoscope
